src/gpodder/plugins/itunes.py

```diff
@@ -27,34 +27,34 @@
 
 logger = logging.getLogger(__name__)
 
-ITUNES_DEFAULT_VERSION = '11.1.5'
-
-ITUNES_FEEDURL_RE = {'10.7': r'feed-url="([^"]+)"',
-                     '11.1.5': r'"feedUrl":\s*"([^"]+)"'}
-
-
 class ITunesFeedException(BaseException):
     pass
 
 
 @registry.feed_handler.register
 def itunes_feed_handler(channel, max_episodes, config):
-    m = re.match(r'https?://(podcasts|itunes).apple.com/(?:[^/]*/)?podcast/.+$', channel.url, re.I)
+    m = re.match(r'https?://(podcasts|itunes)\.apple\.com/(?:[^/]*/)?podcast/.*id(?P<podcast_id>[0-9]+).*$', channel.url, re.I)
     if m is None:
         return None
 
     logger.debug('Detected iTunes feed.')
-    version = ITUNES_DEFAULT_VERSION
-    headers = {'User-agent': 'iTunes/{}'.format(version)}
+
+    if m.group('podcast_id').isnumeric == False or int(m.group('podcast_id')) < 0:
```

Comment:
```suggestion
    if not m.group('podcast_id').isnumeric or int(m.group('podcast_id')) < 0:
```

Reply:
Alternatively, just regex-match for `\d+`?

Reply:
The regex match should not allow a negative number or a non-number into that variable, I'm just very used to double checking that a variable is what it is supposed to be before using it.

Reply:
`[0-9]+`

Reply:
@elelay correct, as said I may just be too paranoid about checking a variable that is generated by regex matching and should only be numeric to begin with.

Reply:
"The regex match should not allow a negative number or a non-number into that variable, I'm just very used to double checking that a variable is what it is supposed to be before using it."

`\d+` does exactly that, it allows the digits `0` to `9` and nothing else (in particular, no `-` or `.`, etc..).

`.isnumeric` is a member function, so if you really want to use it, you should at least call it...